packages/knip/src/plugins/angular/index.ts

```diff
@@ -61,6 +62,34 @@ const resolveConfig: ResolveConfig<AngularCLIWorkspaceConfiguration> = async (co
           }
         }
       }
+      if (target.builder === '@angular-devkit/build-angular:karma' && opts) {
+        const karmaBuilderOptions = opts as KarmaTarget;
+        // https://github.com/angular/angular-cli/blob/19.0.6/packages/angular_devkit/build_angular/src/builders/karma/schema.json#L143
+        const testFilePatterns = karmaBuilderOptions.include ?? ['**/*.spec.ts'];
+        for (const testFilePattern of testFilePatterns) {
+          inputs.add(toEntry(testFilePattern));
+        }
+        // https://github.com/angular/angular-cli/blob/19.0.6/packages/angular_devkit/build_angular/src/builders/karma/schema.json#L146
+        const excludedTestFilePatterns = karmaBuilderOptions.exclude ?? [];
+        for (const excludedTestFilePattern of excludedTestFilePatterns) {
+          inputs.add(toEntry(`!${excludedTestFilePattern}`));
+        }
+        const karmaConfig = karmaBuilderOptions.karmaConfig;
+        if (!karmaConfig) {
+          // Hardcoded default Karma config from Angular builder
+          // https://github.com/angular/angular-cli/blob/19.0.6/packages/angular_devkit/build_angular/src/builders/karma/index.ts#L115
+          karma
+            .inputsFromPlugins(
+              ['karma-jasmine', 'karma-chrome-launcher', 'karma-jasmine-html-reporter', 'karma-coverage'],
+              options.manifest.devDependencies
+            )
+            .forEach(inputs.add, inputs);
```

Comment:
Don't like the `.forEach(inputs.add, inputs)`. But [doesn't work](https://stackoverflow.com/a/50882116/3263250) without the latest `inputs` argument to specify `this`. And a for loop (as `forEach` is forbidden by Biome linter) looks too much 🤷  happy to change if someone else is unhappy too

Reply:
Agreed, this is fine really